mri/reconstructors/calibrationless.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+##########################################################################
+# pySAP - Copyright (C) CEA, 2017 - 2018
+# Distributed under the terms of the CeCILL-B license, as published by
+# the CEA-CNRS-INRIA. Refer to the LICENSE file or to
+# http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html
+# for details.
+##########################################################################
+
+"""
+This implements calibrationless reconstruction with different proximities
+"""
+# System import
+import warnings
+
+from .base import ReconstructorBase
+from ..operators import GradAnalysis, GradSynthesis, WaveletN
+
+# Third party import
+from modopt.opt.proximity import SparseThreshold
+from modopt.opt.linear import Identity
+
+
+class SparseCalibrationlessReconstructor(ReconstructorBase):
+    """ This class implements a calibrationless reconstruction based on the
+    L1-norm regularization.
+    For the Analysis case finds the solution for x of:
+        (1/2) * sum(||F x_l - yl||^2_2, n_coils) +
```

Comment:
it's `y_l` and not `yl`
Also since you allow the user to choose the reconstruction you should rename the function to `CalibrationlessReconstructor` instead